benchmarks/test/test_discovery.py

```diff
@@ -0,0 +1,103 @@
+                  response_size=1024,
+                  concurrency=1):
+  if hasattr(fixture, "proxy_server"):
+    assert (fixture.proxy_server.enableCpuProfiler())
+  assert (fixture.test_server.enableCpuProfiler())
+  args = [
+      fixture.getTestServerRootUri(), "--rps",
+      str(rps), "--duration",
+      str(duration), "--connections",
+      str(max_connections), "--max-active-requests",
+      str(max_active_requests), "--concurrency",
+      str(concurrency), "--request-header",
+      "x-nighthawk-test-server-config:{response_body_size:%s}" % response_size,
+      "--experimental-h1-connection-reuse-strategy", "lru", "--prefetch-connections"
+  ]
+  if request_body_size > 0:
+    args.append("--request-body-size")
+    args.append(str(request_body_size))
+
+  parsed_json, _ = fixture.runNighthawkClient(args)
+  counters = fixture.getNighthawkCounterMapFromJson(parsed_json)
+  response_count = counters["benchmark.http_2xx"]
+  request_count = counters["upstream_rq_total"]
+  connection_counter = "upstream_cx_http1_total"
+
+  # Some arbitrary sanity checks
+  assertCounterGreaterEqual(counters, "benchmark.http_2xx", (rps * duration) * 0.99)
+  assertGreater(counters["upstream_cx_rx_bytes_total"], response_count * response_size)
+  assertGreater(counters["upstream_cx_tx_bytes_total"], request_count * request_body_size)
+  assertCounterEqual(counters, connection_counter, max_connections)
+
+  # Could potentially set thresholds on acceptable latency here.
+
+  # dump human readable output to logs
+  json_as_string = json.dumps(parsed_json)
+  human_output = fixture.transformNighthawkJson(json_as_string, "human")
+  logging.info(human_output)
+
+  with open(os.path.join(fixture.test_server.tmpdir, "nighthawk-human.txt"), "w") as f:
+    f.write(human_output)
+  with open(os.path.join(fixture.test_server.tmpdir, "nighthawk.json"), "w") as f:
+    f.write(json_as_string)
+  with open(os.path.join(fixture.test_server.tmpdir, "nighthawk.yaml"), "w") as f:
+    f.write(fixture.transformNighthawkJson(json_as_string, "yaml"))
+  with open(os.path.join(fixture.test_server.tmpdir, "fortio.json"), "w") as f:
+    f.write(fixture.transformNighthawkJson(json_as_string, "fortio"))
+  with open(os.path.join(fixture.test_server.tmpdir, "server_version.txt"), "w") as f:
+    f.write(fixture.test_server.getCliVersionString())
+  if hasattr(fixture, "proxy_server"):
+    with open(os.path.join(fixture.test_server.tmpdir, "proxy_version.txt"), "w") as f:
+      f.write(fixture.proxy_server.getCliVersionString())
+  r = runfiles.Create()
+  with open(r.Rlocation("nighthawk/benchmarks/test/templates/simple_plot.html"), "r") as r:
+    txt = r.readlines()
+    with open(os.path.join(fixture.test_server.tmpdir, "simple_plot.html"), "w") as w:
+      # This will source nighthawk.json over http from the same dir.
+      # TODO(oschaaf): consider injecting the json directly into a html script source.
+      w.write(str.join("", txt))
```

Comment:
Are you just trying to copy the file here? Why not use direct utils for this such as `shutil.copyfile`?

Reply:
I anticipated we'd might want to run replacements here, but so far we don't, so yeah, followed up and using `shutil.copyfile` now.